--- a/build.zig
+++ b/build.zig
@@ -813,6 +813,8 @@ fn addCmakeCfgOptionsToExe(
                     else => |e| return e,
                 };
                 mod.linkSystemLibrary("unwind", .{});
+                mod.linkSystemLibrary("pthread", .{});
+                mod.linkSystemLibrary("dl", .{});
             },
             .ios, .macos, .watchos, .tvos, .visionos => {
                 mod.link_libcpp = true;
